api/type.rb

```diff
@@ -192,6 +192,19 @@ def initialize(name)
       NAME = Api::Type::String.new('name')
     end
 
+    # Represents a fingerprint.  A fingerprint is an output-only
+    # field used for optimistic locking during updates.
+    class Fingerprint < String
+      def validate
+        super
+        @output = true if @output.nil?
+        # Fingerprints are not usually worth including in the
+        # list of output fields.  This should be overridden in
+        # Terraform, where outputting fingerprints is idiomatic.
+        @exclude ||= true
```

Comment:
The comment confuses me a little bit. It seems like the comment goes along with line 200.

As for making `exclude = true` the default, that's temporary, right? Could you put a TODO or something to mark that as temporary?

Reply:
I suppose it is temporary!  We have to work out how fingerprints ought to behave in puppet, chef, ansible, which means we need custom updates in those.  Added the TODO.